crates/wasi-common/tokio/src/sched/windows.rs

```diff
@@ -0,0 +1,248 @@
+use crate::asyncify;
```

Comment:
This seems like a pretty intense file that has a helper thread to work with stdio on Windows, could this get commented with a high level overview of how all this works?

(I'm not actually sure off the top of my head why this is necessary)

Reply:
This code is borrowed from cap-std-sync which is in turn borrowed from the old version of wasi-common. I also don't quite understand it, and this code was highly resistant to some earlier attempts to improve it, so I have just been dragging it along. I can add better comments, though.